.github/workflows/buildBackendImage.yml

```diff
@@ -0,0 +1,46 @@
+name: Build backend docker image
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "backend/build.gradle"
+      - "backend/settings.gradle"
+      - "backend/gradle.properties"
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}/backend
+
+jobs:
+  build-and-push-backend-image:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
```

Comment:
I'm sure I'm missing something obvious but how does the action know that it's building the backend? I don't see a working directory anywhere.

Reply:
I was in fact missing something obvious: the path comes from the build `context`. In each of these actions, the `context` is `.`, meaning that all 3 actions are trying to build the same `Dockerfile`.

Reply:
Ooof, you are correct. Good catch. I'll fix that!